integrations/llamaindex/src/unitycatalog_ai_llamaindex/toolkit.py

```diff
@@ -0,0 +1,172 @@
+    """
+
+    client_config: Dict[str, Any] = Field(
+        description="Configuration of the client for managing the tool",
+    )
+
+    def __init__(
+        self, fn: Callable, metadata: ToolMetadata, client_config: Dict[str, Any], *args, **kwargs
+    ):
+        """
+        Initializes the UnityCatalogTool.
+
+        Args:
+            fn (Callable): The function that represents the tool's functionality.
+            metadata (ToolMetadata): Metadata about the tool, including name, description, and schema.
+            client_config (Dict[str, Any]): Configuration dictionary for the client used to manage the tool.
+            *args: Additional positional arguments.
+            **kwargs: Additional keyword arguments.
+        """
+        super().__init__(fn=fn, metadata=metadata, *args, **kwargs)
+        self.client_config = client_config
+
+
+class LlamaIndexToolkit(BaseModel):
+    """
+    A toolkit for managing Unity Catalog functions and converting them into tools.
+
+    Attributes:
+        function_names (List[str]): List of function names in 'catalog.schema.function' format.
+        tools_dict (Dict[str, FunctionTool]): A dictionary mapping function names to their corresponding tools.
+        client (Optional[BaseFunctionClient]): The client used to manage functions.
+        return_direct (bool): Whether the tool should return the output directly.
+    """
+
+    function_names: List[str] = Field(
+        default_factory=list,
+        description="List of function names in 'catalog.schema.function' format",
+    )
+    tools_dict: Dict[str, FunctionTool] = Field(default_factory=dict)
+    client: Optional[BaseFunctionClient] = Field(
+        default=None,
+        description="Client for managing functions",
+    )
+    return_direct: bool = Field(
+        default=False,
+        description="Whether the tool should return the output directly",
+    )
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    @model_validator(mode="after")
+    def validate_toolkit(self) -> "LlamaIndexToolkit":
+        """
+        Validates the toolkit configuration and processes function names.
+
+        Returns:
+            LlamaIndexToolkit: The validated and updated toolkit instance.
+        """
+        client = validate_or_set_default_client(self.client)
+        self.client = client
+
+        function_names = self.function_names
+        tools_dict = self.tools_dict
+
+        self.tools_dict = process_function_names(
+            function_names=function_names,
+            tools_dict=tools_dict,
+            client=client,
+            uc_function_to_tool_func=self.uc_function_to_llama_tool,
+        )
+        return self
+
+    @classmethod
+    def uc_function_to_llama_tool(
+        cls,
+        *,
+        client: Optional[BaseFunctionClient] = None,
+        function_name: Optional[str] = None,
+        function_info: Optional[Any] = None,
+        return_direct: Optional[bool] = False,
```

Comment:
Question: when can we pass this parameter? process_function_names doesn't handle this param

Reply:
ah, good catch - I forgot to update the passthrough logic using kwargs.